src/data/set/basic.lean

```diff
@@ -1,25 +1,86 @@
 /-
 Copyright (c) 2014 Jeremy Avigad. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
-Author: Jeremy Avigad, Leonardo de Moura
+Authors: Jeremy Avigad, Leonardo de Moura
 -/
+
 import tactic.basic tactic.finish data.subtype logic.unique
-open function
 
-/-! # Basic properties of sets
+/-!
+
+# Basic properties of sets
+
+Sets in Lean are homogeneous; all their elements have the same type. Sets whose elements
+have type `X` are thus defined as `set X := X → Prop`. The definition is in the core library.
+
+This file provides some basic definitions related to sets and functions not present in the core
+library, as well as extra lemmas for functions in the core library (empty set, univ, union,
+intersection, insert, singleton, set-theoretic difference, complement, and powerset).
+
+Note that a set is a term, not a type. There is a coersion from `set α` to `Type*` sending
+`s` to the corresponding subtype `↑s`.
+
+See also the file `set_theory/zfc.lean`, which contains an encoding of ZFC set theory in Lean.
+
+## Main definitions
+
+Notation used here:
+
+  `f : α → β` is a function,
+  `s : set α` and `s₁ s₂ : set α` are subsets of `α`
```

Comment:
```suggestion
  `s : set α` and `s₁ s₂ : set α` are subsets of `α`,
```
Note also that these lines without bullet points most likely won't render on separate lines in the HTML pages: https://observablehq.com/@bryangingechen/github-lean-doc-preview?url=%221991%22